### ISSUE TITLE
Parsing issue in ergo core

### ISSUE THREAD

**Issue by Tigzy** [evilsocket/ergo#5]
Hello, 
Seems like there's a parsing issue in serialize_classification_report (utils.py)
When (like in my case) the accuracy has no value for precision and recall, it leads to crash when trying to serialize to json. The text file is properly displayed.

I don't think counting spaces is a good idea, however it could be a temporary fix.

![image](https://user-images.githubusercontent.com/9531438/58461310-a6006300-812f-11e9-9775-d44c7132ec21.png)

**Comment by Tigzy** [comment]
So I've inserted the following line at line 30 (original file, not the capture above): 
`if len(row) > len(measures):` 
And if you don't care about the json report it seems to be fixing it. 

![image](https://user-images.githubusercontent.com/9531438/58466387-dcdb7680-8139-11e9-9727-d5dab8248a2f.png)

Side note: That'd be great to mention you also need to install **python3-tk** package somewhere in the doc

**Comment by evilsocket** [comment]
could you attach the dataset so I can reproduce it?

Besides:

![Schermata da 2019-05-28 11-19-50](https://user-images.githubusercontent.com/86922/58466719-8884c680-813a-11e9-81b5-c9624e4f4fcd.png)

**Comment by Tigzy** [comment]
Ok for the doc, thanks. 
The dataset is yours, from the pe-av example.

**Comment by evilsocket** [comment]
I just loaded the dataset with no issue whatsoever, maybe some more details about your config? can you post here the output of `ergo info` ?

**Comment by Tigzy** [comment]
Sure. EDIT: Can you tell me what versions of tf/keras you have ?
Also Ubuntu 16.04, Python 3.5.2

![image](https://user-images.githubusercontent.com/9531438/58467168-7192a400-813b-11e9-9cf5-f8476d7ba550.png)

**Comment by evilsocket** [comment]
Can you try:

    sudo pip3 uninstall tensorflow
    sudo pip3 install tensorflow==1.12.0

**Comment by Tigzy** [comment]
Ok, done. I'm restarting a training.

**Comment by nicochidt** [comment]
Hi Tigzy, 

It's very odd that you don't have those metrics in the report. That's generated from sklearn. Can you please report the output of this code snippet: 

```{python}
from sklearn.metrics import classification_report
from sklearn import __version__
print (__version__)
a = [ 0,1,0,1]
b = [0,1,1,0]
print (classification_report(a,b))
```

**Comment by Tigzy** [comment]
tf to 1.12 didn't fix, let me get this output @nicochidt

**Comment by evilsocket** [comment]
to be honest i'm glad it didn't, it'd be very f weird

**Comment by Tigzy** [comment]
Here you go (don't mind my monkey typing :p) 
It's not there either...

![image](https://user-images.githubusercontent.com/9531438/58468502-f2529f80-813d-11e9-84cf-4753b49d193f.png)

**Comment by evilsocket** [comment]
this is what you should be seeing ...

![Schermata da 2019-05-28 11-45-14](https://user-images.githubusercontent.com/86922/58468571-14e4b880-813e-11e9-8fad-c6d56b74db2f.png)

**Comment by Tigzy** [comment]
Maybe I should revert to version 0.20 ? Is that a python module ?

**Comment by Tigzy** [comment]
Wtf 0.0 version.... :/ 

![image](https://user-images.githubusercontent.com/9531438/58468795-87ee2f00-813e-11e9-8f15-5bafdd3d44c1.png)

**Comment by evilsocket** [comment]
oook, so the problem is caused by sklearn, the new version broke backwards compatibility by removing the `micro avg` metric and replacing it with `accuracy` which has less columns in the classification_report ... working on a fix for both

ironically, it was *my* environment to be outdated, apologies.

**Comment by Tigzy** [comment]
Thanks a lot ! So in theory if I want to continue working on it today I just need to revert sklearn to 0.20 right ?

**Comment by Tigzy** [comment]
Oh I see there's a fix already, neat.

**Comment by evilsocket** [comment]
or just uninstall `ergo-ai` and install this repo from sources

**Comment by Tigzy** [comment]
yeah yeah I was heading this way rather.

**Comment by Tigzy** [comment]
Was looking at the fix and shouldn't it be a label replacement somewhere as well? Where's that accuracy VS micro avg defined?

**Comment by evilsocket** [comment]
i don't understand the question ... the new version just does not present any micro average, but instead there's a new global accuracy metric

**Comment by Tigzy** [comment]
The question is: I see there's None returned when unable to parse properly
Isn't that possible to have actual accuracy (the old micro avg) values instead of just handle the parsing issue and ignore rows ? 

![image](https://user-images.githubusercontent.com/9531438/58474082-888cc280-814a-11e9-8442-34cbd673d401.png)

**Comment by Tigzy** [comment]
Oh OK I think I understand. The new accuracy doesn't replace micro AVG, it's a totally other metric that only has f1-score and support, am I right ?

**Comment by evilsocket** [comment]
yes
